stdlib/Random/test/runtests.jl

```diff
@@ -406,15 +429,12 @@ for rng in ([], [MersenneTwister(0)], [RandomDevice()], [Xoshiro()])
     rand!(rng..., BitMatrix(undef, 2, 3))  ::BitArray{2}
 
     # Test that you cannot call randn or randexp with non-Float types.
-    for r in [randn, randexp, randn!, randexp!]
-        local r
+    for r in [randn, randexp]
```

Comment:
I didn't follow why you removed `randn!` and `randexp!` here.

Reply:
Because those tests seemed like obvious mistakes.

The test claims to be checking *"that you cannot call randn or randexp with non-Float types"*, but in that context it makes no sense to test whether `randn!(T)` is a `MethodError`.    The error in that case has nothing to do with `T` — it is throwing because you are not passing an array to `randn!`

Reply:
(I noticed this because `randn(AbstractFloat)` changed to an `ArgumentError` until I changed the error thrown by `Sampler(...)`.   But of course `rand!` was still throwing a `MethodError`.)

Reply:
Yes ok makes sense.